homeassistant/components/mqtt/cover.py

```diff
@@ -111,31 +110,41 @@ def validate_options(value):
     """
     if CONF_SET_POSITION_TOPIC in value and CONF_GET_POSITION_TOPIC not in value:
         raise vol.Invalid(
-            "'set_position_topic' must be set together with 'position_topic'."
+            f"'{CONF_SET_POSITION_TOPIC}' must be set together with '{CONF_GET_POSITION_TOPIC}'."
         )
 
-    if (
-        CONF_GET_POSITION_TOPIC in value
-        and CONF_STATE_TOPIC not in value
-        and CONF_VALUE_TEMPLATE in value
-    ):
-        _LOGGER.warning(
-            "Using 'value_template' for 'position_topic' is deprecated "
-            "and will be removed from Home Assistant in version 2021.6, "
-            "please replace it with 'position_template'"
+    # if templates are set make sure the topic for the template is also set
+
+    if CONF_VALUE_TEMPLATE in value and CONF_STATE_TOPIC not in value:
+        raise vol.Invalid(
+            f"'{CONF_VALUE_TEMPLATE}' must be set together with '{CONF_STATE_TOPIC}'."
+        )
+
+    if CONF_GET_POSITION_TEMPLATE in value and CONF_GET_POSITION_TOPIC not in value:
+        raise vol.Invalid(
+            f"'{CONF_GET_POSITION_TEMPLATE}' must be set together with '{CONF_GET_POSITION_TOPIC}'."
         )
 
-    if CONF_TILT_INVERT_STATE in value:
-        _LOGGER.warning(
-            "'tilt_invert_state' is deprecated "
-            "and will be removed from Home Assistant in version 2021.6, "
-            "please invert tilt using 'tilt_min' & 'tilt_max'"
+    if CONF_SET_POSITION_TEMPLATE in value and CONF_SET_POSITION_TOPIC not in value:
+        raise vol.Invalid(
+            f"'{CONF_SET_POSITION_TEMPLATE}' must be set together with '{CONF_SET_POSITION_TOPIC}'."
+        )
+
+    if CONF_TILT_COMMAND_TEMPLATE in value and CONF_TILT_COMMAND_TOPIC not in value:
+        raise vol.Invalid(
+            f"'{CONF_TILT_COMMAND_TEMPLATE}' must be set together with '{CONF_TILT_COMMAND_TOPIC}'."
+        )
+
+    if CONF_TILT_STATUS_TEMPLATE in value and CONF_TILT_STATUS_TOPIC not in value:
+        raise vol.Invalid(
+            f"'{CONF_TILT_STATUS_TEMPLATE}' must be set together with '{CONF_TILT_STATUS_TOPIC}'."
         )
 
     return value
 
 
 PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_TILT_INVERT_STATE),
```

Comment:
I'm not sure this is right because `CONF_TILT_INVERT_STATE` no longer does anything, which means a configuration which sets `CONF_TILT_INVERT_STATE = True` will no longer work as intended.
I think it's better to either just drop `CONF_TILT_INVERT_STATE` altogether and fail when it's used, or warn if `CONF_TILT_INVERT_STATE = False` and fail if `CONF_TILT_INVERT_STATE = True`?

Reply:
I think the best option is to drop `CONF_TILT_INVERT_STATE` and let it fail when it is used as it is already showing a warning for few releases. When it is set to `True` config must fail and since this is a not a mandatory option I assume it will be mostly used when it is set to `True`.